chunks 1,2: Always set .Nm in bold, including in NAME (local regression fix).
chunk 3: Set .Sx in italics, not in roman with quotes (local regression fix).

Index: tmac/doc.tmac
--- tmac/doc.tmac.orig
+++ tmac/doc.tmac
@@ -1186,11 +1186,6 @@
 .  if !\n[doc-arg-count] \{\
 .    ds doc-macro-name Nm
 .    ie \n[.$] \{\
-.      \" Handle '.Nm ...' in "Name" section.
-.      if \n[doc-in-name-section] \{\
-.        if "\*[doc-topic-name]"" \
-.          ds doc-topic-name "\$1\"
-.      \}
 .      doc-parse-args \$@
 .    \}
 .    el \{\
@@ -1248,6 +1243,8 @@
 .          \}
 .          ti -\n[doc-indent-synopsis]u
 .      \}\}
+.      if "\*[doc-topic-name]"" \
+.        ds doc-topic-name "\*[doc-arg\n[doc-arg-ptr]]
 .
 .      nop \f[\*[doc-Nm-font]]\c
 .    \}
@@ -4785,23 +4782,9 @@ arguments
 .\" NS   cross section reference
 .\" NS
 .\" NS width register 'Sx' set in doc-common
-.\"
-.\" TODO: This duplicates the definition of `Dq`; figure out how to
-.\" simply wrap that macro if possible.  (It's not trivial to do so
-.\" because of mdoc's design feature of recursively calling macro
-.\" arguments as macros.)
 .
-.eo
-.de Sx
-.  if !\n[doc-arg-count] \
-.    ds doc-macro-name Sx
-.
-.  ds doc-quote-left "\*[Lq]
-.  ds doc-quote-right "\*[Rq]
-.
-.  doc-enclose-string \$@
-..
-.ec
+.als Sx doc-generic-macro
+.ds doc-Sx-usage section_header
 .
 .
 .\" NS doc-end-column-list macro
